src/openforms/registrations/contrib/objects_api/submission_registration.py

```diff
@@ -339,8 +361,34 @@ class ObjectsAPIV2Handler(ObjectsAPIRegistrationHandler[RegistrationOptionsV2]):
     def get_object_data(
         self, submission: Submission, options: RegistrationOptionsV2
     ) -> dict[str, Any]:
+
         state = submission.load_submission_value_variables_state()
         dynamic_values = state.get_data()
+
+        # For every file upload component, we alter the value of the variable to be
+        # the Document API URL(s).
+        objects_api_attachments = ObjectsAPISubmissionAttachment.objects.filter(
+            submission_file_attachment__submission_variable__submission=submission
+        ).annotate(
+            variable_key=F("submission_file_attachment__submission_variable__key")
+        )
+
+        urls_map: defaultdict[str, list[str]] = defaultdict(list)
+        for o in objects_api_attachments:
+            urls_map[o.variable_key].append(o.document_url)
+
+        for key in dynamic_values.keys():
+            if key in urls_map:
+                variable = state.get_variable(key)
+                is_multiple = variable.form_variable.form_definition.configuration_wrapper.component_map[
+                    key
+                ].get(
+                    "multiple", False
+                )
+                dynamic_values[key] = (
+                    urls_map[key][0] if not is_multiple else urls_map[key]
+                )
```

Comment:
I did some reasoning about the `form_variable` FK potentially being `None` and how this works if registration is run after a file-upload component is removed from the form, and I concluded it's not an issue, because:

* `state` is used, which ensures that we only consider currently available form variables (through the `collect_variables` method)
* `dynamic_values` is used, which relies on the above state considering only currently available form variables

So there may be extra `SubmissionFileAttachment` records for the submission, and those will exist in our DB but not sent along to the objects API. And this behaviour is as (currently) intended.